synapse/app/synctl.py

```diff
@@ -76,8 +79,14 @@ def start_worker(app, configfile, worker_configfile):
 def stop(pidfile, app):
     if os.path.exists(pidfile):
         pid = int(open(pidfile).read())
-        os.kill(pid, signal.SIGTERM)
-        write("stopped %s" % (app,), colour=GREEN)
+        try:
+            os.kill(pid, signal.SIGTERM)
+            write("stopped %s" % (app,), colour=GREEN)
+        except OSError, err:
+            if err.errno == errno.ESRCH:
+                write("%s not running" % (app,), colour=YELLOW)
+            else:
+                write("Cannot stop %s" % (app,), colour=RED)
```

Comment:
(a) this doesn't tell us *why* it couldn't stop it
(b) it will then carry on with the rest of the script rather than aborting